Automattic/jetpack
# Jetpack Recommendations: Fix existing Product Suggestions tracks

#### Introduction



This PR aims to correct tracking events related to the newly introduced Product Suggestion flow in the Jetpack Assistant. 

* Origin PR: #20814 

#### Changes proposed in this Pull Request:


Three of the new tracks had implementation issues:

* `jetpack_recommendations_product_suggestion_click`: due to passing a string instead of object with properties.
* `jetpack_recommendations_product_suggestion_learn_more_click`: due to passing a string instead of object with properties.
* `jetpack_recommendations_product_suggestion_purchased`: due to incorrectly defining the purchased product.

#### Jetpack product discussion



This PR aims to fix tracking of an existing features introduced in the latest 10.1 release and there is no direct conversation/P2 about it. There is however an existing PT which contains a comment that briefly addresses the issue.

* PT: pbNhbs-Yp-p2#comment-4074

#### Does this pull request change what data or activity we track or use?



Technically, no, because the events did not show up in our system, so these can be considered "new" but have previously been mentioned and made aware about in the Jetpack division.

#### Testing instructions:






1. Spin up a site using this branch.
2. Create a Jetpack Free connection.
3. Go to `/wp-admin/admin.php?page=jetpack#/recommendations/`
4. Fill out the site type (the selection here doesn't matter)

Ones you reach the Product Suggestions screen you need to do the following (in chronological order):

1. Press any of the `Learn More` links to trigger a "external link event".
2. Close the new tab that opened (so you end up back at the WordPress website again).
3. Press any of the `Continue with <PRODUCT>` buttons, located above the "External link".
4. Purchase the product in the checkout. This can be done with credit without affecting the end result - we just need a purchase so you will be redirected back to the website.
5. After purchasing a product/plan you should be redirect to `/wp-admin/admin.php?page=jetpack#/recommendations/product-purchased` which is the "Confirmation page" where we tell them what product they've purchased.

When you've done the above, you should then have triggered all 3 tracking events mentioned in the description of this PR.
If you wait ~10min you should be able to validate that the data was registered in our live tracking tool (PCYsg-5hv-p2) and optionally filter by your WPCOM username  - even though there will probably not be that many events, so just look for your own username to make sure it's the ones you triggered.
